util.c

```diff
@@ -201,3 +201,75 @@ int r_mkdir(char *path)
 	}
 	return 0;
 }
+
+void construct_argv(char **argv, unsigned int len, ...)
+{
+	unsigned int i;
+	va_list args;
+
+	va_start(args, len);
+	for (i = 0; i < len; ++i)
+		argv[i] = va_arg(args, char *);
+	va_end(args);
+	if (argv[len-1] != NULL)
+		error(EXIT_FAILURE, 0, "argv not NULL terminated");
+}
+
+static void xclose(int fd)
+{
+	if (fd >= 0)
+		close(fd);
+}
```

Comment:
This seems like a code smell. We shouldn't be calling close on an invalid fd.

Also think it is easier to read if we just call close directly. `close`'s behavior is well known, but someone would have to explicitly what xclose does.